app/src/main/res/layout/item_watchlist.xml

```diff
@@ -0,0 +1,130 @@
+            android:textSize="16sp"
+            android:textColor="?attr/material_theme_secondary_color"
+            tools:text="12:00"/>
+
+        <ImageView
+            android:id="@+id/langCodeBackground"
+            android:layout_width="0dp"
+            android:layout_height="18dp"
+            android:contentDescription="@null"
+            app:layout_constraintBottom_toBottomOf="@+id/langCodeText"
+            app:layout_constraintEnd_toEndOf="@+id/langCodeText"
+            app:layout_constraintStart_toStartOf="@+id/langCodeText"
+            app:layout_constraintTop_toTopOf="@+id/langCodeText"
+            app:srcCompat="@drawable/lang_button_shape_border"
+            app:tint="?attr/material_theme_secondary_color" />
+
+        <TextView
+            android:id="@+id/langCodeText"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintTop_toTopOf="@id/timeText"
+            app:layout_constraintBottom_toBottomOf="@id/timeText"
+            app:layout_constraintEnd_toStartOf="@id/timeText"
+            android:layout_marginEnd="8dp"
+            android:fontFamily="monospace"
+            android:lineSpacingExtra="2sp"
+            android:padding="3dp"
+            android:paddingEnd="4dp"
+            android:textAllCaps="true"
+            android:textColor="?attr/material_theme_secondary_color"
+            android:textSize="10sp"
+            android:textStyle="bold"
+            tools:text="zh-hant" />
+
+        <TextView
+            android:id="@+id/summaryText"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            app:layout_constraintTop_toBottomOf="@id/titleText"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            android:layout_marginTop="6dp"
+            android:textSize="16sp"
+            android:lineSpacingExtra="8sp"
+            android:textColor="?attr/material_theme_secondary_color"
+            android:textColorLink="?attr/material_theme_secondary_color"
+            tools:text="Lorem ipsum"/>
+
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/diffText"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            style="@style/FlatButton.Small"
+            android:layout_marginTop="4dp"
+            android:paddingTop="8dp"
+            android:paddingBottom="8dp"
+            android:paddingStart="12dp"
+            android:paddingEnd="12dp"
+            app:layout_constraintTop_toBottomOf="@id/summaryText"
+            app:layout_constraintStart_toStartOf="parent"
+            android:fontFamily="sans-serif-medium"
+            android:textAllCaps="false"
+            android:textColor="@color/green50"
+            tools:text="+123"/>
+
+        <com.google.android.material.button.MaterialButton
```

Comment:
The IPv6 address does not fit the screen properly.
![Screenshot_20201208-134621_Wikipedia Dev](https://user-images.githubusercontent.com/2435576/101546539-c84a8200-395d-11eb-9423-783e716beb26.jpg)

Reply:
Yep, I just gave it a maxWidth, which should take care of these cases.